rrtmgp/mo_gas_optics_rrtmgp.F90

```diff
@@ -1516,10 +1539,12 @@ function compute_optimal_angles(this, optical_props, optimal_angles) result(err_
     integer  :: ncol, nlay, ngpt
     integer  :: icol, ilay, igpt, bnd
     real(wp) :: t, trans_total
+    integer, allocatable :: bands(:)
```

Comment:
Maybe `integer :: bands(optical_props%get_ngpt())` to avoid explicit allocation ?

Reply:
`integer :: bands(optical_props%get_ngpt())` was failing with the CCE compiler, that is why it has to be allocated explicitly, at least for the time being.